pkg/features/features.go

```diff
@@ -0,0 +1,231 @@
+				f.mux.Lock()
+				// Set the supported features
+				f.features = featuresMap
+				f.mux.Unlock()
+				// Stop polling and close this channel
+				f.featureRequestTicker.Stop()
+			}
+		}
+	}()
+}
+
+// Stop stops the request ticker
+func (f *Features) Stop() {
+	f.featureRequestTicker.Stop()
+}
+
+// getSupportedFeatures returns the features supported by the StackState API
+func (f *Features) getSupportedFeatures() {
+	f.mux.Lock()
+	// Lock so only one goroutine at a time can access the map
+	f.retriesLeft = f.retriesLeft - 1
+	if f.retriesLeft == 0 {
+		f.featureChan <- map[string]bool{}
+	}
+	f.mux.Unlock()
+
+	resp, accessErr := f.makeFeatureRequest()
+	// Handle error response
+	if accessErr != nil {
+		// Soo we got a 404, meaning we were able to contact stackstate, but it had no features path. We can publish a result
+		if resp != nil {
+			log.Info("Found StackState version which does not support feature detection yet")
+			return
+		}
+		// Log
+		_ = log.Error(accessErr)
+		return
+	}
+
+	defer resp.Body.Close()
+
+	// Get byte array
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		_ = log.Errorf("could not decode response body from features: %s", err)
+		return
+	}
+	var data interface{}
+	// Parse json
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		_ = log.Errorf("error unmarshalling features json: %s of body %s", err, body)
+		return
+	}
+
+	// Validate structure
+	featureMap, ok := data.(map[string]interface{})
+	if !ok {
+		_ = log.Errorf("Json was wrongly formatted, expected map type, got: %s", reflect.TypeOf(data))
+		return
+	}
+
+	featuresParsed := make(map[string]bool)
+
+	for k, v := range featureMap {
+		featureValue, okV := v.(bool)
+		if !okV {
+			_ = log.Warnf("Json was wrongly formatted, expected boolean type, got: %s, skipping feature %s", reflect.TypeOf(v), k)
+		}
+		featuresParsed[k] = featureValue
+	}
+
+	log.Infof("Server supports features: %v", featuresParsed)
+	f.featureChan <- featuresParsed
+}
+
+// GetRetriesLeft returns the retry count
+func (f *Features) GetRetriesLeft() int {
+	f.mux.Lock()
+	// Lock so only one goroutine at a time can access the map
+	defer f.mux.Unlock()
+	return f.retriesLeft
+}
+
+// FeatureEnabled checks whether a certain feature is enabled
+func (f *Features) FeatureEnabled(feature string) bool {
+	f.mux.Lock()
+	// Lock so only one goroutine at a time can access the map
+	defer f.mux.Unlock()
+	if supported, ok := f.features[feature]; ok {
+		return supported
+	}
+	return false
+}
+
+func (f *Features) makeFeatureRequest() (*http.Response, error) {
+	url := fmt.Sprintf("%s/features", f.endpoint.Host)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not create request to %s/features: %s", url, err)
+	}
+
+	req.Header.Add("content-encoding", "identity")
+	req.Header.Add("sts-api-key", f.endpoint.APIKey)
+	req.Header.Add("sts-hostname", f.endpoint.Host)
+	req.Header.Add("sts-traceagentversion", info.Version)
```

Comment:
the features fetching can be eventually used by the all agent right? not only by the trace one
I would change the user agent in `sts-agent-version`

Reply:
Ah yes, good catch